server/src/main/java/org/apache/iotdb/db/metadata/mtree/store/disk/cache/CacheMemoryManager.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.metadata.mtree.store.disk.cache;
+
+import org.apache.iotdb.db.metadata.mtree.store.CachedMTreeStore;
+import org.apache.iotdb.db.metadata.mtree.store.disk.MTreeFlushTaskManager;
+import org.apache.iotdb.db.metadata.mtree.store.disk.MTreeReleaseTaskManager;
+import org.apache.iotdb.db.metadata.mtree.store.disk.memcontrol.IMemManager;
+import org.apache.iotdb.db.metadata.mtree.store.disk.memcontrol.MemManagerHolder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CacheMemoryManager {
+
+  private final List<CachedMTreeStore> storeList = new ArrayList<>();
+
+  private final IMemManager memManager = MemManagerHolder.getMemManagerInstance();
+
+  private final MTreeFlushTaskManager flushTaskManager = MTreeFlushTaskManager.getInstance();
+  private int flushCount = 0;
+  private volatile boolean hasFlushTask;
+
+  private final MTreeReleaseTaskManager releaseTaskManager = MTreeReleaseTaskManager.getInstance();
+  private volatile boolean hasReleaseTask;
+  private int releaseCount = 0;
+
+  public ICacheManager createLRUCacheManager(CachedMTreeStore store) {
+    ICacheManager cacheManager = new LRUCacheManager();
+    storeList.add(store);
+    return cacheManager;
+  }
+
+  public void ensureMemoryStatus() {
+    if (memManager.isExceedFlushThreshold() && !hasReleaseTask) {
+      registerReleaseTask();
+    }
+  }
+
+  private synchronized void registerReleaseTask() {
+    if (hasReleaseTask) {
+      return;
+    }
+    hasReleaseTask = true;
+    releaseTaskManager.submit(this::tryExecuteMemoryRelease);
```

Comment:
It seems there won' be more than two release tasks being executed concurrently. The efficiency is not much high enough, and it's a waste to prepare an independent thread pool for such tasks.

Reply:
replaced by SingleThreadExecutor

Reply:
Use CompleteFuture to release and flush concurrently.